CxxHighlighter is requesting multiple highlights for overlapping section/lines

When analyzing files within my C++ project, multiple highlighters have been requested for the same section of text within the source.  It seems SonarQube will check for overlapping highlight ranges and throw errors when it finds overlapping highlight sections.

Here is the line that SonarQube is complaining about

```
        qWarning() << "Failed to open file " << sourceFilename << " for reading";
```

And here is the exception

```
org.sonar.squidbridge.api.AnalysisException: Unable to analyze file: File.cpp
	at org.sonar.squidbridge.AstScanner.scanFiles(AstScanner.java:131)
	at org.sonar.plugins.cxx.squid.CxxSquidSensor.execute(CxxSquidSensor.java:128)
	at org.sonar.scanner.sensor.SensorWrapper.analyse(SensorWrapper.java:53)
	at org.sonar.scanner.phases.SensorsExecutor.executeSensor(SensorsExecutor.java:57)
	at org.sonar.scanner.phases.SensorsExecutor.execute(SensorsExecutor.java:49)
	at org.sonar.scanner.phases.AbstractPhaseExecutor.execute(AbstractPhaseExecutor.java:78)
	at org.sonar.scanner.scan.ModuleScanContainer.doAfterStart(ModuleScanContainer.java:182)
	at org.sonar.core.platform.ComponentContainer.startComponents(ComponentContainer.java:142)
	at org.sonar.core.platform.ComponentContainer.execute(ComponentContainer.java:127)
	at org.sonar.scanner.scan.ProjectScanContainer.scan(ProjectScanContainer.java:247)
	at org.sonar.scanner.scan.ProjectScanContainer.scanRecursively(ProjectScanContainer.java:242)
	at org.sonar.scanner.scan.ProjectScanContainer.doAfterStart(ProjectScanContainer.java:232)
	at org.sonar.core.platform.ComponentContainer.startComponents(ComponentContainer.java:142)
	at org.sonar.core.platform.ComponentContainer.execute(ComponentContainer.java:127)
	at org.sonar.scanner.task.ScanTask.execute(ScanTask.java:47)
	at org.sonar.scanner.task.TaskContainer.doAfterStart(TaskContainer.java:86)
	at org.sonar.core.platform.ComponentContainer.startComponents(ComponentContainer.java:142)
	at org.sonar.core.platform.ComponentContainer.execute(ComponentContainer.java:127)
	at org.sonar.scanner.bootstrap.GlobalContainer.executeTask(GlobalContainer.java:115)
	at org.sonar.batch.bootstrapper.Batch.executeTask(Batch.java:118)
	at org.sonarsource.scanner.api.internal.batch.BatchIsolatedLauncher.execute(BatchIsolatedLauncher.java:62)
	at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
	at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.lang.reflect.Method.invoke(Method.java:498)
	at org.sonarsource.scanner.api.internal.IsolatedLauncherProxy.invoke(IsolatedLauncherProxy.java:60)
	at com.sun.proxy.$Proxy0.execute(Unknown Source)
	at org.sonarsource.scanner.api.EmbeddedScanner.doExecute(EmbeddedScanner.java:233)
	at org.sonarsource.scanner.api.EmbeddedScanner.runAnalysis(EmbeddedScanner.java:151)
	at org.sonarsource.scanner.cli.Main.runAnalysis(Main.java:110)
	at org.sonarsource.scanner.cli.Main.execute(Main.java:74)
	at org.sonarsource.scanner.cli.Main.main(Main.java:61)
Caused by: java.lang.IllegalStateException: Cannot register highlighting rule for characters at Range[from [line=163, lineOffset=38] to [line=163, lineOffset=46]] as it overlaps at least one existing rule
	at org.sonar.api.batch.sensor.highlighting.internal.DefaultHighlighting.checkOverlappingBoudaries(DefaultHighlighting.java:59)
	at org.sonar.api.batch.sensor.highlighting.internal.DefaultHighlighting.doSave(DefaultHighlighting.java:122)
	at org.sonar.api.batch.sensor.internal.DefaultStorable.save(DefaultStorable.java:43)
	at org.sonar.plugins.cxx.highlighter.CxxHighlighter.leaveFile(CxxHighlighter.java:118)
	at com.sonar.sslr.impl.ast.AstWalker.walkAndVisit(AstWalker.java:71)
	at org.sonar.squidbridge.AstScanner.scanFiles(AstScanner.java:110)
	... 31 more
```

--- a/sonar-cxx-plugin/src/test/resources/org/sonar/plugins/cxx/highlighter.cc
+++ b/sonar-cxx-plugin/src/test/resources/org/sonar/plugins/cxx/highlighter.cc
@@ -73,4 +73,9 @@ void test1()
    const std::regex RegexEscape(R"([.^$|()\[\]{}*+?\\])"); // raw string literal
 }
 
+void test2(const char* sourceFilename)
+{
+   Warning() << "Failed to open file " << sourceFilename << " for reading";
+}
+
 /* EOF */

